test/replication-luatest/gh_5295_split_brain_test.lua

```diff
@@ -0,0 +1,247 @@
+local t = require('luatest')
```

Comment:
I've got this when run the test 30 times:
```
python test-run.py $(yes 5295 | head -n 30)
```
```
[012] replication-luatest/gh_5295_split_brain_test.l>                 [ fail ]
[012] Test failed! Output from reject file /tmp/t/rejects/replication-luatest/gh_5295_split_brain.reject:
[012] TAP version 13
[012] 1..8
[012] # Started on Wed Jun 15 00:10:05 2022
[012] # Starting group: gh-6036
[012] not ok 1	gh-6036.test_async_old_term
[012] #   Peer closed
[012] #   stack traceback:
[012] #   	builtin/box/net_box.lua:747: in function '_request'
[012] #   	builtin/box/net_box.lua:807: in function 'eval'
[012] #   	...ol/test/replication-luatest/gh_5295_split_brain_test.lua:95: in function 'partition_nth_instance'
[012] #   	...ol/test/replication-luatest/gh_5295_split_brain_test.lua:136: in function 'gh-6036.test_async_old_term'
[012] #   	...
[012] #   	[C]: in function 'xpcall'
[012] s1 | Assertion failed: (replicaset.applier.synced > 0), function replica_on_applier_disconnect, file /Users/gerold/Work/Repositories/tarantool/src/box/replication.cc, line 498.
[012] not ok 2	gh-6036.test_confirm_old_term
```
Is it related to the patch? I didn't see you changing `synced` counter or anything related in this PR.

Reply:
Yep, I've seen it too. I don't know how it could be connected to the patch.
This might be some uncovered issue with starting a too big cluster.

On my side the issue disappeared when I removed the `replication_connect_quorum = 1` line from the node config.
Looks like I've simply decreased the chances to reproduce it.

I think this will disappear once I rework the test with only having 2 nodes in cluster at all times.

Reply:
Lets pretend it never happened so we don't need to file a ticket.

Reply:
Yep. Looks like the issue reproduces when bootstrapping a rather big cluster. Say, 9 nodes. Like it was with previous test version.